pkg/processor/trigger/kafka/trigger.go

```diff
@@ -205,6 +205,11 @@ func (k *kafka) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.C
 	workerDrainingCompleteChan := make(chan bool)
 	readyForRebalanceChan := make(chan bool)
 
+	// indicate whether this partition worker was drained
+	// this is used to avoid race condition where 2 different partitions sharing the same worker
+	// will both try to reset the drain flag needlessly
+	var drainedWorker bool
```

Comment:
Maybe it's better to use atomic bool here? Just to be sure that we don't write and read and the same time from both goroutines.